semantic-analysis/src/Analysis/Syntax.hs

```diff
@@ -79,6 +82,8 @@ eval eval = \case
     v' <- eval v
     let' n v' (eval b)
   Import ns -> S.simport ns >> dunit
+  Function n ps b -> letrec n (dabs ps (\ as ->
+    foldr (\ (p, a) m -> let' p a m) (eval b) (zip ps as)))
```

Comment:
Functions are treated as definitions (i.e. not lambdas), so we bind them recursively. In future we'll want to distinguish between definitions and their values, and between recursive and non-recursive bindings (i.e. `letrec` and `let`), but here again, it's good enough for demo work.